src/org/jetbrains/plugins/scala/caches/HydraArtifactsCache.scala

```diff
@@ -0,0 +1,43 @@
+package org.jetbrains.plugins.scala.caches
+
+import java.util.concurrent.ConcurrentHashMap
+
+import com.intellij.openapi.diagnostic.Logger
+import org.jetbrains.plugins.scala.project.Platform
+import org.jetbrains.plugins.scala.project.template.Downloader
+
+/**
+  * @author Maris Alexandru
+  */
+object HydraArtifactsCache {
+  private val LOG = Logger.getInstance(this.getClass)
+  private val cache = new ConcurrentHashMap[(String, String), Seq[String]]()
+  private val SplitRegex = "\\[info\\] \\* Attributed\\(|\\)"
+  private val ArtifactsRegex = "\\[info\\] \\* Attributed\\(.*\\)"
+  private val HydraRegex = ".*hydra_(\\d+\\.\\d+\\.\\d+)-(\\d+\\.\\d+\\.\\d+(-SNAPSHOT)?)\\.jar".r
+
+  private def cacheArtifacts(artifacts: String): Unit = {
+    val paths = artifacts.split("\n").filter(s => s.matches(ArtifactsRegex)).map(s => s.split(SplitRegex)(1))
```

Comment:
Writing `s.matches(ArtifactsRegex)` results in the `ArtifactsRegex` to be compiled several times. If you add `.r` to the definition of `ArtifactsRegex` (see my above comment), then the regular expression will be compiled. To use the compiled regular expression you can replace `s.matches(ArtifactsRegex)` with `ArtifactsRegex.findFirstIn(s).nonEmpty` or `ArtifactsRegex.pattern.matcher(text).matches()` ( I wonder why there is no `matches` method in the Scala Regex library to make this a bit nicer :))

By the way, this comment is valid for every occurrence where you are matching a regex

Reply:
Furthermore, I think we should have some trace in the LOG if the match fails (again, the comment is valid in general)

Reply:
Thanks for the explanation, Mirco. This is very helpful!